app/Controllers/MediaController.php

```diff
@@ -188,6 +188,50 @@ public function download(Request $request, Response $response, string $userCode,
         return $this->streamMedia($request, $response, $this->storage, $media, 'attachment');
     }
 
+
+    /**
+     * @param  Request  $request
+     * @param  Response  $response
+     * @param  string  $vanity
+     * @param  string  $id
+     *
+     * @return Response
+     * @throws HttpNotFoundException
+     * @throws HttpBadRequestException
+     */
+    public function createVanity(Request $request, Response $response, int $id): Response
+    {
+        if (!$this->session->get('admin')) {
+            $media = $this->database->query('SELECT * FROM `uploads` WHERE `id` = ? LIMIT 1', $id)->fetch();
+        } else {
+            $media = $this->database->query(
+                'SELECT * FROM `uploads` WHERE `id` = ? AND `user_id` = ? LIMIT 1',
+                [$id, $this->session->get('user_id')]
+            )->fetch();
+        }
+
+        $data = $request->getParsedBody();
+        $vanity = $data['vanity'];
+        $vanity = strtolower(preg_replace('/[^a-z0-9]+/', '-', $vanity));
+
+        if (!$media) {
+            throw new HttpNotFoundException($request);
+        }
+
+        if ($vanity === '' || $media->code === $vanity) {
+            throw new HttpBadRequestException($request);
+        }
+
+        $this->database->query(
+            'UPDATE `uploads` SET `code` = ? WHERE `id` = ?',
+            [$vanity, $media->id]
+        );
+
+        $this->logger->info('User '.$this->session->get('username').' created a vanity link for media '.$media->id);
+        
+        return $response;
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** ‏ ‏what about returning the updated media object here? So the frontend can update the HTML accordingly

Reply:
You mean to like this `return $media;`? Thats because it is not a valid Psr\Http\Message\ResponseInterface type which slim expects to be returned. Otherwise, it will give an 500 error

Reply:
Nevermind, I figured out how to return the media object. Turns out you need to add `$response->getBody()->write(json_encode($media));`, for it to return data. Thought unfortunately, it does not work with an onject so I had to convert it first.